ts/indicators/ema.src.ts

```diff
@@ -0,0 +1,230 @@
+        }
+
+    }
+}
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType,
+    correctFloat = H.correctFloat;
+
+/**
+ * The EMA series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.ema
+ *
+ * @augments Highcharts.Series
+ */
+seriesType<Highcharts.EMAIndicator>(
+    'ema',
+    'sma',
+    /**
+     * Exponential moving average indicator (EMA). This series requires the
+     * `linkedTo` option to be set.
+     *
+     * @sample stock/indicators/ema
+     *         Exponential moving average indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @optionparent plotOptions.ema
+     */
+    {
+        params: {
+            /**
+             * The point index which indicator calculations will base. For
+             * example using OHLC data, index=2 means the indicator will be
+             * calculated using Low values.
+             *
+             * By default index value used to be set to 0. Since Highstock 7
+             * by default index is set to 3 which means that the ema
+             * indicator will be calculated using Close values.
+             */
+            index: 3,
+            period: 9 // @merge 14 in v6.2
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        accumulatePeriodPoints: function (
+            period: number,
+            index: number,
+            yVal: Array<Array<number>>
+        ): number {
+            var sum = 0,
+                i = 0,
+                y: (number|Array<number>) = 0;
+
+            while (i < period) {
+                y = index < 0 ? yVal[i] : yVal[i][index];
+                sum = sum + (y as any);
+                i++;
+            }
+
+            return sum;
+        },
+        calculateEma: function (
+            xVal: (Array<number>|undefined),
+            yVal: Array<Array<number>>,
+            i: number,
+            EMApercent: number,
+            calEMA: (number|undefined),
+            index: number,
+            SMA: number
+        ): [number, number] {
+            var x: number = (xVal as any)[i - 1],
+                yValue: (number|Array<number>) = index < 0 ?
+                    yVal[i - 1] :
+                    yVal[i - 1][index],
+                y: number;
+
+            y = calEMA === undefined ?
+                SMA : correctFloat(((yValue as any) * EMApercent) +
+                (calEMA * (1 - EMApercent)));
+
+            return [x, y];
+        },
+        getValues: function (
+            this: Highcharts.EMAIndicator,
+            series: Highcharts.EMAIndicator,
+            params: Highcharts.EMAIndicatorParamsOptions
+        ): (boolean|Highcharts.IndicatorValuesObject) {
+            var period: number = (params.period as any),
+                xVal: (Array<number>|undefined) = series.xData,
+                yVal: Array<Array<number>> = series.yData,
```

Comment:
You have to make use of `as any` on the right side for both value arrays. Additionally `xVal` should not be undefinable.